tests/src/ExistingSiteJavascript/CollaboraIntegrationTest.php

```diff
@@ -0,0 +1,121 @@
+        // expected text or values.
+        $this->waitForRequiredElement('css', 'canvas#document-canvas', 2000);
+        // Make sure the correct document was opened.
+        // Check the document name at the top of the editor.
+        $this->assertSame(
+            $text_file_basename . '.txt',
+            // Even when the element exists, it might not have the correct value
+            // yet.
+            $this->waitForRequiredElement('css', 'input#document-name-input')->waitFor(
+                1000,
+                fn (Element $element) => $element->getValue(),
+            ),
+        );
+        // The document text is in a canvas element, so instead we compare the
+        // word count and character count.
+        $this->assertSame(
+            '2 words, 18 characters',
+            $this->waitForRequiredElement('css', 'div#StateWordCount')->waitFor(
+                1000,
+                fn (Element $element) => $element->getText(),
+            ),
+            $assert_session->elementExists('css', 'div#StateWordCount')->getText(),
+        );
+    }
+
+    /**
+     * Creates a media entity with an attached *.txt file.
+     *
+     * The *.txt format is enough to test the basic functionality.
+     *
+     * @param string $media_name
+     *   Media label.
+     * @param string $file_basename
+     *   File name without the extension.
+     * @param string $text_content
+     *   Content for the attached *.txt file.
+     *
+     * @return \Drupal\media\MediaInterface
+     *   New media entity.
+     */
+    protected function createDocumentMedia(string $media_name, string $file_basename, string $text_content): MediaInterface {
+        $file_uri = 'public://' . $file_basename . '.txt';
+        file_put_contents($file_uri, $text_content);
+        $file = File::create([
+            'uri' => $file_uri,
+        ]);
+        $file->save();
+        $this->markEntityForCleanup($file);
+        $values = [
+            'bundle' => 'document',
+            'name' => $media_name,
+            'title' => $media_name,
+            'label' => $media_name,
+            'field_media_file' => $file->id(),
+        ];
+        $media = Media::create($values);
+        $media->save();
+        $this->markEntityForCleanup($media);
+        return $media;
+    }
+
+    /**
+     * Waits for an element to appear, and asserts its existence.
+     *
+     * @param string $selector
+     *   Selector type, e.g. 'css'.
+     * @param string $locator
+     *   Selector string.
+     * @param string $timeout
+     *   (Optional) Timeout in milliseconds, defaults to 10000.
+     *
+     * @return \Behat\Mink\Element\NodeElement
+     *   The element found for the selector.
+     */
+    protected function waitForRequiredElement($selector, $locator, $timeout = 10000): NodeElement {
+        $this->assertSession()->waitForElement($selector, $locator, $timeout);
+        return $this->assertSession()->elementExists($selector, $locator);
```

Comment:
Instead of pinging again the browser to retrieve the element, just use the return value of the previous call.
```
$element = $this->assertSession()->waitForElement($selector, $locator, $timeout);
$this->assertNotNull($element, sprintf('Could not find element with %s, %s.', $selector, $locator);
return $element;